hibernate-core/src/main/java/org/hibernate/query/hql/internal/SemanticQueryBuilder.java

```diff
@@ -1629,7 +1651,11 @@ public SqmSortSpecification visitSortSpecification(HqlParser.SortSpecificationCo
 
 	@Override
 	public SqmExpression<?> visitSortExpression(HqlParser.SortExpressionContext ctx) {
-		return resolveOrderByOrGroupByExpression( ctx.getChild( 0 ), ctx.getChildCount() > 1 );
+		return visitSortExpression( ctx, true );
+	}
+
+	public SqmExpression<?> visitSortExpression(HqlParser.SortExpressionContext ctx, boolean allowPositionalOrAliases) {
+		return resolveOrderByOrGroupByExpression( ctx.getChild( 0 ), ctx.getChildCount() > 1, allowPositionalOrAliases );
```

Comment:
I updated the PR with @beikov's suggestions, disallowing alias/positional based order by in OVER and WITHIN GROUP clauses.

Impl. note: I had to pass the flag down to `visitSortExpression` because calling `ctx.getChild( 0 ).accept( this )` was not enough to resolve the order-by expression. The logic is now managed in `resolveOrderByOrGroupByExpression`.